biz.aQute.bnd/src/aQute/bnd/main/ResolveCommand.java

```diff
@@ -268,28 +272,33 @@ public void _resolve(ResolveOptions options) throws Exception {
 			if (!f.isFile()) {
 				error("Missing bndrun file: %s", f);
 			} else {
-
 				Run run = Run.createRun(ws, f);
-
-				try (ProjectResolver pr = new ProjectResolver(run)) {
-					try {
-						Map<Resource,List<Wire>> resolution = pr.resolve();
-						if (pr.isOk()) {
-							System.out.printf("# %-50s ok\n", f.getName());
-							if (options.bundles()) {
-								for (Resource r : resolution.keySet()) {
-									IdentityCapability id = ResourceUtils.getIdentityCapability(r);
-									List<ContentCapability> content = ResourceUtils.getContentCapabilities(r);
-									System.out.printf("  %-50s %40s %s\n", id.osgi_identity(),
-											content.get(0).osgi_content(), content.get(0).url());
+				if (options.write()) {
+					Bndrun bndrun = new Bndrun(run.getWorkspace(), f);
+					String resolved = bndrun.resolve(false, true);
```

Comment:
? Bndrun does exactly the same as the current body. Since Bndrun itself does not do any output, you can keep the current body and, as BJ seems to suggest, just use write as a flag to resolve?Just refactoring the old body and use Bndrun seems the best solution or am I missing something?